Cannot find module 'feathers-reactive'.

Hi guys,

I'm trying to use feathers reactive in a angular cli project, all other feather modules are imported fine, except from feathers-reactive.

I keep getting Cannot find module 'feathers-reactive'., when I'm compiling.
I'm using latest version of feathers-reactive and I'm using angular/cli": "1.0.0-beta.32.3

I can see that the feathers-reactive folder is correctly in node_modules.

Is there any other way to import it? I'm doing this currently: import * as reactive from 'feathers-reactive';

Best regards
Patrick

The first gives me this error:
ERROR in /Users/patrickpereira/lazy/backend/src/app/shared/api/api.service.ts (11,22): Cannot find module 'feathers-reactive'.)

Second one is working, so I'm just using that.

Still failing, steps to reproduce:

1) Download https://github.com/feathersjs/feathers-chat-angular
2) Update feathers-reactive version to 0.5.0 in package.json
3) run: `npm install`
4) run: `ng serve`

** Note ** I tried this but didn't work: Try adding `declare module 'feathers-reactive'` to `node_modules/feathers-reactive/index.d.ts`

Result:

```
ERROR in ./src/app/services/feathers.service.ts
Module not found: Error: Can't resolve 'feathers-reactive' in '/Users/ohull/Downloads/feathers-chat-angular-master/src/app/services'
 @ ./src/app/services/feathers.service.ts 12:0-48
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
```

--- a/package.json
+++ b/package.json
@@ -3,7 +3,7 @@
   "description": "Reactive API extensions for Feathers services",
   "version": "0.5.0",
   "homepage": "https://github.com/feathersjs/feathers-reactive",
-  "main": "src/index",
+  "main": "lib/index",
   "keywords": [
     "feathers",
     "feathers-plugin"

--- a/src/index.js
+++ b/src/index.js
@@ -1,26 +1,25 @@
-const { matcher } = require('feathers-commons/lib/utils');
-const reactiveResource = require('./resource');
-const reactiveList = require('./list');
-const strategies = require('./strategies');
-const {
-  makeSorter,
-  getParamsPosition
-} = require('./utils');
+import _debug from 'debug';
+import { matcher } from 'feathers-commons/lib/utils';
+import { Observable } from 'rxjs/Observable';
 
-const { Observable } = require('rxjs/Observable');
-require('rxjs/add/observable/fromEvent');
-require('rxjs/add/observable/fromPromise');
-require('rxjs/add/observable/merge');
-require('rxjs/add/operator/concat');
-require('rxjs/add/operator/exhaustMap');
-require('rxjs/add/operator/filter');
-require('rxjs/add/operator/let');
-require('rxjs/add/operator/map');
-require('rxjs/add/operator/mapTo');
-require('rxjs/add/operator/mergeMap');
-require('rxjs/add/operator/scan');
+import 'rxjs/add/observable/fromEvent';
+import 'rxjs/add/observable/fromPromise';
+import 'rxjs/add/observable/merge';
+import 'rxjs/add/operator/concat';
+import 'rxjs/add/operator/exhaustMap';
+import 'rxjs/add/operator/filter';
+import 'rxjs/add/operator/let';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/mapTo';
+import 'rxjs/add/operator/mergeMap';
+import 'rxjs/add/operator/scan';
 
-const debug = require('debug')('feathers-reactive');
+import reactiveResource from './resource';
+import reactiveList from './list';
+import strategies from './strategies';
+import { makeSorter, getParamsPosition } from './utils';
+
+const debug = _debug('feathers-reactive');
 
 function FeathersRx (options = {}) {
   const listStrategies = strategies();

--- a/src/list.js
+++ b/src/list.js
@@ -1,12 +1,12 @@
-const {
+import 'rxjs/add/operator/finally';
+import 'rxjs/add/operator/shareReplay';
+
+import {
   getOptions,
   getSource,
   cacheObservable,
   getCachedObservable
-} = require('./utils');
-
-require('rxjs/add/operator/finally');
-require('rxjs/add/operator/shareReplay');
+} from './utils';
 
 module.exports = function (settings) {
   return function (params) {

--- a/src/resource.js
+++ b/src/resource.js
@@ -1,11 +1,11 @@
-const {
+import { Observable } from 'rxjs/Observable';
+import {
   getOptions,
   getSource,
   cacheObservable,
   getCachedObservable,
   getParamsPosition
-} = require('./utils');
-const { Observable } = require('rxjs/Observable');
+} from './utils';
 
 module.exports = function (settings, method) {
   return function () {

--- a/src/strategies.js
+++ b/src/strategies.js
@@ -1,4 +1,4 @@
-const { Observable } = require('rxjs/Observable');
+import { Observable } from 'rxjs/Observable';
 
 module.exports = function () {
   return {

--- a/src/utils.js
+++ b/src/utils.js
@@ -1,7 +1,9 @@
-const createSorter = require('feathers-commons/lib/utils').sorter;
-const { Observable } = require('rxjs/Observable');
-const stringify = require('json-stable-stringify');
-const debug = require('debug')('feathers-reactive');
+import _debug from 'debug';
+import { sorter as createSorter } from 'feathers-commons/lib/utils';
+import { Observable } from 'rxjs/Observable';
+import stringify from 'json-stable-stringify';
+
+const debug = _debug('feathers-reactive');
 
 function getSource (originalMethod, args) {
   let resultPromise = null;

